### PR TITLE
Get favorite by id with /favorites/:id endpoint

### LINKED ISSUE #8: User can get a specific favorite with /favorites/:id endpoint
User sends specific favorite id via params `GET /api/v1/favorites/:id`

User can expect a successful response body as follows with a `200 status` with a favorite object: 
```
{
  "id": 1,
  "title": "We Will Rock You",
  "artistName": "Queen"
  "genre": "Rock",
  "rating": 88
}
```
Return `404 status` if favorite object is not found

Acceptance criteria: 
- [ ] All tests passing 
- [ ] Happy path
- [ ] Sad path 
- [ ] Peer review
- [ ] Deployed to Heroku

### PR DESCRIPTION
**Pull request by lrs8810**
## Type of change 

- [ ] :beetle: Bug fix (non-breaking change which fixes an issue)
- [x] :hatching_chick: New feature (non-breaking change which adds functionality)
- [x] :page_with_curl: This change requires a documentation update

## Summary of changes 
- Add `getFavoriteByID.spec.js` file in tests directory testing `GET /api/v1/favorites/:id` happy path, sad paths resulting in 404 and 500
- Add route `GET /api/v1/favorites/:id` to `favorites.js` file

## How Has This Been Tested?

- [x] :black_square_button: Integration testing 
- [ ] :white_square_button: Unit testing 

## Checklist:

- [x] :white_check_mark: I have performed a self-review of my own code
- [x] :eight_spoked_asterisk: I have commented my code, particularly in hard-to-understand areas
- [ ] :page_facing_up: I have made corresponding changes to the documentation
- [x] :no_entry_sign: My changes generate no new warnings
- [x] :heavy_check_mark: I have added tests that prove my fix is effective or that my feature works
- [x] :100: New and existing unit tests pass locally with my changes

### CODE REVIEW

File: routes/api/v1/favorites.js [hunk 1]
```diff
@@ -41,4 +41,22 @@ router.post('/', (request, response) => {
       });
 });
 
+router.get('/:id', (request, response) => {
+  let id = request.params.id
+  database('favorites').where('id', id)
+  .select('id', 'title', 'artistName', 'genre', 'rating')
+  .then(favorite => {
+    if (favorite.length) {
+      response.status(200).json(favorite[0]);
```
Comment (lrs8810) [inline routes/api/v1/favorites.js]: Do you think we should assign this to `favorite[0]` to a variable and then call it here?

File: tests/getFavoriteByID.spec.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    it('sad path, will return 404 if favorite is not found', async () => {
+      const res = await request(app)
+        .get("/api/v1/favorites/30")
+        expect(res.statusCode).toEqual(404);
+        expect(res.body).toHaveProperty("error");
+        expect(res.body.error).toBe("Could not find favorite with id 30");
+    });
+    
+    it('sad path, will return 500 if :id is anything other than an integer', async () => {
```
Comment (lrs8810) [inline tests/getFavoriteByID.spec.js]: Please let me know if this sad path documentation is confusing

File: tests/getFavoriteByID.spec.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      const res = await request(app)
+        .get("/api/v1/favorites/start")
+        expect(res.statusCode).toEqual(500);
+        expect(res.body).toHaveProperty("error");
```
Comment (lrs8810) [inline tests/getFavoriteByID.spec.js]: Should we have an `expect` for `res.body.error`? The only thing is I don't know what the returning error message will be. Do you think we should have a specific message returned?

File: routes/api/v1/favorites.js (hunk 1 above)
Comment (zacisaacson) [reply routes/api/v1/favorites.js]: Ya we could definitely do that!

File: tests/getFavoriteByID.spec.js (hunk 2 above)
Comment (zacisaacson) [reply tests/getFavoriteByID.spec.js]: It looks great to me

File: tests/getFavoriteByID.spec.js (hunk 3 above)
Comment (zacisaacson) [reply tests/getFavoriteByID.spec.js]: I'm not exactly sure if we should have a specific message every time, might be a good thing to ask the instructors

**Pull request merged**